src/lib/codec-wrappers/codec.ts

```diff
@@ -1,5 +1,5 @@
 export interface Encoder {
-  encode(data: ImageData): Promise<ArrayBuffer>;
+  encode(data: ImageData, options: any): Promise<ArrayBuffer>;
```

Comment:
Jake and I discussed in the original PR as well and I said I’d like to keep the `encode()` function consistent across all codecs and have individual methods for each option.

Reply:
I'm happy for it to land as-is, and we can have a discussion when we start to integrate options.

Reply:
I wasn't sure what we wanted the lifecycle of these encoder classes to be. If we reuse one instance for multiple encodes, methods are pain.  Also given the fact that state is managed elsewhere, duplicating it here just causes extra work.